views/calendar/calendar.js

```diff
@@ -130,8 +124,9 @@ window.calendarView = class calendarView extends View {
                         }
                     });
                 }
+
                 // Then we insert the content into the correct month
-                // This could be simplified...
+                // Mapping does not appear to be working here...
                 for (var i = 0; i < x.length; i++) {
                     try {
                         $(`#Jan`).append(`
```

Comment:
@harrislineage are we talking about this part of code?

```js
$(`#Jan`).append(`
<span>${AllMonths.Jan[i].day} ${getMonth(AllMonths.Jan[i].date)}
${AllMonths.Jan[i].year} <a href="https://www.wikitree.com/wiki/${AllMonths.Jan[i].id}">
${AllMonths.Jan[i].name}</a> ${AllMonths.Jan[i].text} ${
    AllMonths.Jan[i].marriage
} [<span class="small"><a href="https://www.wikitree.com/treewidget/${
    AllMonths.Jan[i].id
}/6">share tree image</a>]
</span><br>
`);

// and further ...
```

If yes, I'm little bit struggling here
- 1st - I'm not able to run it locally - I don't know why, but `watchlist`  returns me empty  array on localhost (not empty in production 🤔 )
- 2nd - what does the array `AllMonths.Jan` / `AllMonths.Feb` / ... contain? Already sorted events per month?

Reply:
If that's the case (events sorted per month) wouldn't this work?

```js
for (monthName of MONTHS) { // loop over all months
    const shortName = monthName.slice(0, 3);
    const events = AllMonths[shortName];

    $(`#${shortName}`).append(
        events // loop over all events in the month
            .map(
                (event) => `
                    <div>
                        ${event.day} ${getMonth(event.date)} ${event.year}
                        <a href="https://www.wikitree.com/wiki/${event.id}">${event.name}</a>
                        ${event.text} ${event.marriage} 
                        [<span class="small">
                            <a href="https://www.wikitree.com/treewidget/${event.id}/6">
                                share tree image
                            </a>
                        </span>]
                    </div>`
            )
            .join("")
    );
}
```